apps/web/src/common/modules/widgets/_widget-fields/date-range/WidgetFieldDateRange.vue

```diff
@@ -0,0 +1,605 @@
+} from '@cloudforet/mirinae';
+import type { MenuItem } from '@cloudforet/mirinae/types/inputs/context-menu/type';
+
+import { useProxyValue } from '@/common/composables/proxy-state';
+import {
+    DATE_RANGE_DAILY_VALUE_MAP,
+    DATE_RANGE_DAILY_VALUES, DATE_RANGE_MONTHLY_VALUE_MAP,
+    DATE_RANGE_MONTHLY_VALUES, DATE_RANGE_YEARLY_VALUE_MAP, DATE_RANGE_YEARLY_VALUES,
+} from '@/common/modules/widgets/_widget-fields/date-range/constant';
+import type { DateRangeOptions, DateRangeValue, DateRangeValueType } from '@/common/modules/widgets/_widget-fields/date-range/type';
+import type {
+    WidgetFieldComponentProps,
+    WidgetFieldComponentEmit,
+} from '@/common/modules/widgets/types/widget-field-type';
+
+const GRANULARITY_UNIT_MAP = {
+    MONTHLY: 'Month',
+    DAILY: 'Day',
+    YEARLY: 'Year',
+};
+
+const MONTHLY_ENABLED_VALUES = ['auto', ...DATE_RANGE_MONTHLY_VALUES, 'custom'];
+const DAILY_ENABLED_VALUES = ['auto', ...DATE_RANGE_DAILY_VALUES, 'custom'];
+const YEARLY_ENABLED_VALUES = ['auto', ...DATE_RANGE_YEARLY_VALUES, 'custom'];
+
+const getCommonDateRangeValueLabel = (value: string): string => {
+    if (value === 'auto') return 'Auto';
+    if (value === 'custom') return 'Custom';
+    return 'Auto';
+};
+
+const props = defineProps<WidgetFieldComponentProps<DateRangeOptions, DateRangeValue>>();
+const emit = defineEmits<WidgetFieldComponentEmit<DateRangeValue>>();
+const state = reactive({
+    granularity: computed<string>(() => (props.allValueMap?.granularity as string|undefined) || 'MONTHLY'),
+    baseDateRange: computed(() => props.dateRange),
+    proxyValue: useProxyValue<DateRangeValue>('value', props, emit),
+    valueMenuItems: computed(() => {
+        if (state.granularity === 'MONTHLY') return MONTHLY_ENABLED_VALUES.map((value) => ({ label: DATE_RANGE_MONTHLY_VALUE_MAP[value] || getCommonDateRangeValueLabel(value), name: value }));
+        if (state.granularity === 'DAILY') return DAILY_ENABLED_VALUES.map((value) => ({ label: DATE_RANGE_DAILY_VALUE_MAP[value] || getCommonDateRangeValueLabel(value), name: value }));
+        if (state.granularity === 'YEARLY') return YEARLY_ENABLED_VALUES.map((value) => ({ label: DATE_RANGE_YEARLY_VALUE_MAP[value] || getCommonDateRangeValueLabel(value), name: value }));
+        return [];
+    }),
+    datePreview: computed<{start: string; end: string;}>(() => {
+        const isInherit = !!state.proxyValue.inherit;
+        const _granularity = state.granularity;
+        const relativeDateRangeValue = state.proxyValue?.options?.value;
+
+        const baseEndDate = isInherit ? state.baseDateRange?.end : undefined;
+        const thisYear = dayjs.utc(baseEndDate).year();
+        const thisMonth = dayjs.utc(baseEndDate).month() + 1;
+        const baseDate = baseEndDate ? dayjs.utc(baseEndDate).endOf('month') : dayjs.utc(baseEndDate);
+
+        // Results
+        let _start = '';
+        let _end = '';
+
+        if (_granularity === 'MONTHLY') {
```

Comment:
todo: Please change all `if` statements to `else if`. Otherwise, all conditions will execute sequentially.